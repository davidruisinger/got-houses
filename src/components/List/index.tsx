import React, { FunctionComponent } from 'react'
import styled from '../_utils/styledComponents'
import mediaQuery from '../_utils/mediaQuery'

// Inspired by https://medium.com/cloudaper/how-to-create-a-flexible-square-grid-with-css-grid-layout-ea48baf038f3
const Wrapper = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-auto-rows: 1fr;
  grid-gap: 1em 1em;

  ${mediaQuery.sm`
    grid-template-columns: repeat(3, 1fr);
  `}
  ${mediaQuery.md`
    grid-template-columns: repeat(4, 1fr);
  `}

  ::before {
    content: '';
    width: 0;
    padding-bottom: 100%;
    grid-row: 1 / 1;
    grid-column: 1 / 1;
  }

  > *:first-child {
    grid-row: 1 / 1;
    grid-column: 1 / 1;
  }
`

const List: FunctionComponent<{}> = ({ children }) => (
  <Wrapper>{children}</Wrapper>
)

export default List
