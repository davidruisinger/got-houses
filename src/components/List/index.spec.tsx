import 'jest-styled-components'
import React from 'react'
import testRenderer from 'react-test-renderer'
import _get from 'lodash.get'
import List from './index'

const tree = testRenderer
  .create(
    <List>
      <span>Test child 1</span>
      <span>Test child 2</span>
    </List>
  )
  .toJSON()

describe('<List />', () => {
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('renders the children', () => {
    expect(_get(tree, 'children')).toHaveLength(2)
    expect(_get(tree, 'children[0].type')).toEqual('span')
    expect(_get(tree, 'children[1].type')).toEqual('span')
    expect(_get(tree, 'children[0].children[0]')).toEqual('Test child 1')
    expect(_get(tree, 'children[1].children[0]')).toEqual('Test child 2')
  })
})
