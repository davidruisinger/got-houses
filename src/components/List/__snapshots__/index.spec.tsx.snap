// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-auto-rows: 1fr;
  grid-gap: 1em 1em;
}

.c0::before {
  content: '';
  width: 0;
  padding-bottom: 100%;
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

.c0 > *:first-child {
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

@media (min-width:36em) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:48em) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

<div
  className="c0"
>
  <span>
    Test child 1
  </span>
  <span>
    Test child 2
  </span>
</div>
`;
